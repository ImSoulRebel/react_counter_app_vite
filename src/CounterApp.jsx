import { useState } from "react";
import PropTypes from "prop-types";

export function CounterApp({ value }) {
  const [counter, setCounter] = useState(value);
  const addNumber = () => {
    setCounter(counter + 1);
    // setCounter((c)=> c+1);
  };
  const substractNumber = () => {
    setCounter(counter - 1);
    // setCounter((c)=> c+1);
  };
  const resetNumber = () => {
    setCounter(value);
    // setCounter((c)=> c+1);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={addNumber}>+1</button>
      <button onClick={substractNumber}>-1</button>
      <button aria-label="button-reset" onClick={resetNumber}>
        reset
      </button>
    </>
  );
}

CounterApp.PropTypes = {
  value: PropTypes.number.isRequired, //Genera error que bloquea codigo
};
