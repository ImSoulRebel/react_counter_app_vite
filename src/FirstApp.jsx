/* eslint-disable react/prop-types */
import PropTypes from 'prop-types';

// Variables y funciones fuera de Functional component para evitar reasignamiento de memoria
// const newMessage = 'Bruce!!';

export const FirstApp = ({title, subtitle, name}) =>{
  return (
    // Fragment, para retornar varios elementos .jsx dentro de un nodo padre
    <> 
      <h1 data-testid="test-title">{title}</h1>
      <p>{subtitle}</p>
      <p>{name}</p>
    </>
  )
}

// Obliga a seguir tipados en React+JScript
FirstApp.PropTypes = {
    title: PropTypes.string.isRequired, //Genera error que bloquea codigo
    subtitle: PropTypes.number,
    name: PropTypes.string,
}

FirstApp.defaultProps = {
  name: 'nombre por defecto',
  title: 'No hay titulo',
  subtitle: 0,
}
