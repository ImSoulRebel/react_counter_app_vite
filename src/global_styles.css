export html,body{
    background-color:  #21232A;
    color:white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 7px;
}

button{
    padding: 15px;
    font-size: 1.3rem;
    margin: 5px;
}