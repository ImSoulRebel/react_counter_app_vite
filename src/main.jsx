import React from 'react';
import ReactDOM from 'react-dom/client';
// import {FirstApp} from './FirstApp';
// import {HelloWorldApp} from './HelloWorldApp'
import './global_styles.css';
import {CounterApp} from './CounterApp';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={23}/>
        {/* < FirstApp title="Hello, i'm Vegeta"/> */}
    </React.StrictMode>
);