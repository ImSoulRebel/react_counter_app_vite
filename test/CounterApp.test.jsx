/**
 * @jest-environment jsdom
 */
import {  fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";


/* eslint-disable no-undef */
describe('Test on <CounterApp/>', () => { 
    const initialValue = 100;

    test('Match with snapshot', () => { 
        const {container} = render(<CounterApp value={ initialValue }/>);
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar el valor inicial a 100', () => { 
        render(<CounterApp value={ initialValue }/>);
        expect(screen.getByText(initialValue)).toBeTruthy();
    });

    test('debe incrementar con el botón +1', () => { 
        render(<CounterApp value={ initialValue }/>);
        fireEvent.click(screen.getByText('+1'));
        expect(screen.getByText('101')).toBeTruthy();
    });

    test('debe decrementar con el botón -1', () => { 
        render(<CounterApp value={ initialValue }/>);
        fireEvent.click(screen.getByText('-1'));
        expect(screen.getByText('99')).toBeTruthy();
    });

    test('debe funcionar  el botón reset', () => { 
        render(<CounterApp value={ initialValue }/>);
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        // fireEvent.click(screen.getByText('reset'));
        fireEvent.click(screen.getByRole('button', {name: 'button-reset'}));
   
        expect(screen.getByText(initialValue)).toBeTruthy();
    });
 });