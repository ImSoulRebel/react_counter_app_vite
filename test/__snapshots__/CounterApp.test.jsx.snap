// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <CounterApp/> Match with snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="button-reset"
  >
    reset
  </button>
</div>
`;
